---
title: "Progression Free Survival Analysis of Signature Genes"
output: html_document
date: '2022-12-20'
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r,warning=FALSE,message=FALSE}
library(survival)
library(survminer)
library(pheatmap)
library(dplyr)
library(tidyverse)
```


```{r}
surv_free <- read.table(file = "./data/prog_free_surv.tsv", sep = "\t", header = TRUE)
surv_free <- surv_free[complete.cases(surv_free), ]
```

# GBM Survival Analysis using RNA Expression data and Signature Genes

```{r}

# Load clinical data
clinical <- read.csv("./data/mmc2_clinical_data.csv")
# clinical_sub <- clinical[,c("case_id","vital_status","path_diag_to_last_contact_days","path_diag_to_death_days")]
clinical_sub <- clinical[,c("case_id", "vital_status")]

# for(i in 1:nrow(clinical_sub)){
#   clinical_sub$time[i] <- ifelse(clinical_sub[i,"vital_status"]=="Living",clinical_sub[i,"path_diag_to_last_contact_days"],clinical_sub[i,"path_diag_to_death_days"])
# }

clinical_sub$time <- NA
for(i in row.names(clinical_sub)){
  for (j in row.names(surv_free)) {
    if(clinical_sub[i, "case_id"] == surv_free[j, "Case"]){
      clinical_sub[i, "time"] <- surv_free[j, "Days.to.Recurrence"]
    }
  }
}


# Load gene expression data of GBM
exp <- read.csv("./data/mmc3_expression.csv")
exp_sub <- exp[,c(2,5:ncol(exp))]
```

```{r}
# Exclude IDH mutant patients
idh_patients <- c("C3L-02041","C3L-02708","C3L-02900","C3L-02984","C3N-00663","C3N-03070")
# Signature genes
signature_rna <- c("COL18A1","COL1A2","COL5A1",
               "COL6A2","COL6A3","CYR61","EMILIN1","IGFBP4",
               "IGFBP6","LTBP3","PODNL1","SNED1","TGFBI",
               "THBS1","ASPN","SPON2","COL22A1")

```

```{r}
plot_RNA_heatmap <- function(input_df,idh_patients,signature){
  rownames(input_df) <- make.names(input_df[,1],unique = T)
  input_df <- input_df[,-1] #108 patients
  input_df <- input_df[rownames(input_df)%in%signature,]
  colnames(input_df) <- gsub("\\.","-",colnames(input_df))
  input_df <- input_df[,!colnames(input_df)%in%idh_patients]
  input_df <- log(input_df)
  return(input_df)
}

```

```{r,fig.width=14}
rna_in <- plot_RNA_heatmap(exp_sub,idh_patients,signature_rna)
exp_h <- pheatmap(rna_in,clustering_distance_rows = "manhattan",clustering_distance_cols  = "manhattan",clustering_method = "ward.D")
exp_h
```


## Survival analysis using clusters

```{r,fig.align='center',fig.height=7}

exp_h_clust <- cbind(t(rna_in), 
                     cluster = cutree(exp_h$tree_row, k = 2))
exp_h_clust <- as.data.frame(exp_h_clust)

exp_h_clust$case_id <- rownames(exp_h_clust)


## Remove rows from exp_h_clust which does not exist in surv_free file
exp_h_clust <- exp_h_clust[(exp_h_clust$case_id %in% surv_free$Case), ]


clust_all <- merge(exp_h_clust,clinical_sub,by="case_id")
clust_all$status1 <- ifelse(clust_all$vital_status=="Deceased",1,0)
fit_exp_clust <- survfit(Surv(time, status1) ~ cluster, data = clust_all)
  ggsurvplot(fit_exp_clust,
           pval = TRUE,
           ncensor.plot=FALSE,
           risk.table.col = "strata", 
           ggtheme = theme_bw(), 
           title = "Signature",
           palette = c("blue", "red2"))


```



## Survival analysis using tiles
```{r}
plot_survival <- function(input_df,clinical_sub,idh_patients,signature,tile){
  rownames(input_df) <- make.names(input_df[,1],unique = T)
  input_df <- input_df[,-1] #108 patients
  input_df <- input_df[rownames(input_df)%in%signature,]
  colnames(input_df) <- gsub("\\.","-",colnames(input_df))
  input_df <- input_df[,!colnames(input_df)%in%idh_patients]
  input_df <- log(input_df)
  input_df <- as.data.frame(t(input_df))
  input_df <- na.omit(input_df)
  input_df$signature <- apply(input_df, 1, mean)
  input_df$case_id <- rownames(input_df)
  input_df <- input_df[,c("signature","case_id")]
  all_df <- merge(input_df,clinical_sub,by="case_id")
  all_df <- all_df %>%
    mutate(quantile = ntile(signature, tile))
  all_df <- all_df[all_df$quantile==1|all_df$quantile==tile,]
  all_df$status1 <- ifelse(all_df$vital_status=="Deceased",1,0)
  return(all_df)
}

```

# Survival analysis comparing 1st vs 2nd tiles:

```{r,fig.align='center',fig.height=7}
# Survival analysis of Signature expression high patients VS. signature exp. low patients
rna_all <- plot_survival(exp_sub,clinical_sub,idh_patients,signature_rna,2)


## Remove rows from rna_all which does not exist in surv_free file
rna_all <- rna_all[(rna_all$case_id %in% surv_free$Case), ]


fit_e2 <- survfit(Surv(time, status1) ~ quantile, data = rna_all)

ggsurvplot(fit_e2,
           pval = TRUE,
           ncensor.plot=FALSE,
           risk.table.col = "strata", 
           ggtheme = theme_bw(), 
           title = "Signature",
           palette = c("blue", "red2"))
```

```{r}

# hazard ratio, exp(coef) = 1.1287
cox <- coxph(Surv(time, status1) ~ quantile, data = rna_all)
cox
```

# Survival analysis comparing 1st vs 3rd tiles:

```{r,fig.align='center',fig.height=7}
# Survival analysis of Signature expression high patients VS. signature exp. low patients
rna_all3 <- plot_survival(exp_sub,clinical_sub,idh_patients,signature_rna,3)

## Remove rows from rna_all3 which does not exist in surv_free file
rna_all3 <- rna_all3[(rna_all3$case_id %in% surv_free$Case), ]


fit_e3 <- survfit(Surv(time, status1) ~ quantile, data = rna_all3)

ggsurvplot(fit_e3,
           pval = TRUE,
           ncensor.plot=FALSE,
           risk.table.col = "strata", 
           ggtheme = theme_bw(), 
           title = "Signature",
           palette = c("blue", "red2"))
```


```{r}

# hazard ratio, exp(coef) = 1.1287
cox3 <- coxph(Surv(time, status1) ~ quantile, data = rna_all3)
cox3
```

# Survival analysis comparing 1st vs 4rd tiles:

```{r,fig.align='center',fig.height=7}
# Survival analysis of Signature expression high patients VS. signature exp. low patients
rna_all4 <- plot_survival(exp_sub,clinical_sub,idh_patients,signature_rna,4)

## Remove rows from rna_all4 which does not exist in surv_free file
rna_all4 <- rna_all4[(rna_all4$case_id %in% surv_free$Case), ]


fit_e4 <- survfit(Surv(time, status1) ~ quantile, data = rna_all4)

ggsurvplot(fit_e4,
           pval = TRUE,
           ncensor.plot=FALSE,
           risk.table.col = "strata", 
           ggtheme = theme_bw(), 
           title = "Signature",
           palette = c("blue", "red2"))
```

```{r}

# hazard ratio, exp(coef) = 1.1287
cox4 <- coxph(Surv(time, status1) ~ quantile, data = rna_all4)
cox4
```

# GBM Survival Analysis using Protein Expression data and Signature Genes

```{r}
# Load protein expression data of GBM
prot <- read.csv("./data/mmc3_proteome.csv")
prot_sub <- prot[,c(1,4:ncol(prot))]
```

```{r}
# Signature genes, exclude "SPON2" and "COL22A1" because of missingness in proteomics data
signature_prot <- c("COL18A1","COL1A2","COL5A1",
                    "COL6A2","COL6A3","CYR61","EMILIN1","IGFBP4",
                    "IGFBP6","LTBP3","PODNL1","SNED1","TGFBI",
                    "THBS1","ASPN")
```


```{r}
plot_prot_heatmap <- function(input_df,idh_patients,signature){
  rownames(input_df) <- make.names(input_df[,1],unique = T)
  input_df <- input_df[,-1] #108 patients
  input_df <- input_df[rownames(input_df)%in%signature,]
  colnames(input_df) <- gsub("\\.","-",colnames(input_df))
  input_df <- input_df[,!colnames(input_df)%in%idh_patients]
  return(input_df)
}
```

```{r,fig.width=14}
prot_in <- plot_prot_heatmap(prot_sub,idh_patients,signature_prot)
prot_h <- pheatmap(prot_in,clustering_distance_rows = "manhattan",clustering_distance_cols  = "manhattan",clustering_method = "ward.D")
prot_h


```
## Survival analysis using tiles for Proteomics data

```{r}

plot_survival_prot <- function(input_df,clinical_sub,idh_patients,signature,tile){
  rownames(input_df) <- make.names(input_df[,1],unique = T)
  input_df <- input_df[,-1] #108 patients
  input_df <- input_df[rownames(input_df)%in%signature,]
  colnames(input_df) <- gsub("\\.","-",colnames(input_df))
  input_df <- input_df[,!colnames(input_df)%in%idh_patients]
  input_df <- as.data.frame(t(input_df))
  input_df <- na.omit(input_df)
  input_df$signature <- apply(input_df, 1, mean)
  input_df$case_id <- rownames(input_df)
  input_df <- input_df[,c("signature","case_id")]
  all_df <- merge(input_df,clinical_sub,by="case_id")
  all_df <- all_df %>%
    mutate(quantile = ntile(signature, tile))
  all_df <- all_df[all_df$quantile==1|all_df$quantile==tile,]
  all_df$status1 <- ifelse(all_df$vital_status=="Deceased",1,0)
  return(all_df)
}


```

# Survival analysis comparing 1st vs 2nd tiles:

```{r,fig.align='center',fig.height=7}
# Survival analysis of Signature protein expression high patients VS. signature exp. low patients
prot_all2 <- plot_survival_prot(prot_sub,clinical_sub,idh_patients,signature_prot,2)

## Remove rows from prot_all2 which does not exist in surv_free file
prot_all2 <- prot_all2[(prot_all2$case_id %in% surv_free$Case), ]


fit_p2 <- survfit(Surv(time, status1) ~ quantile, data = prot_all2)

ggsurvplot(fit_p2,
           pval = TRUE,
           ncensor.plot=FALSE,
           risk.table.col = "strata", 
           ggtheme = theme_bw(), 
           title = "Signature",
           palette = c("blue", "red2"))


```

```{r}

# hazard ratio, exp(coef) 
coxp2 <- coxph(Surv(time, status1) ~ quantile, data = prot_all2)
coxp2
```

# Survival analysis comparing 1st vs 3rd tiles:

```{r,fig.align='center',fig.height=7}
# Survival analysis of Signature protein expression high patients VS. signature exp. low patients
prot_all3 <- plot_survival_prot(prot_sub,clinical_sub,idh_patients,signature_prot,3)

## Remove rows from prot_all3 which does not exist in surv_free file
prot_all3 <- prot_all3[(prot_all3$case_id %in% surv_free$Case), ]


fit_p3 <- survfit(Surv(time, status1) ~ quantile, data = prot_all3)

ggsurvplot(fit_p3,
           pval = TRUE,
           ncensor.plot=FALSE,
           risk.table.col = "strata", 
           ggtheme = theme_bw(), 
           title = "Signature",
           palette = c("blue", "red2"))


```

```{r}

# hazard ratio, exp(coef) 
coxp3 <- coxph(Surv(time, status1) ~ quantile, data = prot_all3)
coxp3
```

# Survival analysis comparing 1st vs 4th tiles:

```{r,fig.align='center',fig.height=7}
# Survival analysis of Signature protein expression high patients VS. signature exp. low patients
prot_all4 <- plot_survival_prot(prot_sub,clinical_sub,idh_patients,signature_prot,4)

## Remove rows from prot_all4 which does not exist in surv_free file
prot_all4 <- prot_all4[(prot_all4$case_id %in% surv_free$Case), ]


fit_p4 <- survfit(Surv(time, status1) ~ quantile, data = prot_all4)

ggsurvplot(fit_p4,
           pval = TRUE,
           ncensor.plot=FALSE,
           risk.table.col = "strata", 
           ggtheme = theme_bw(), 
           title = "Signature",
           palette = c("blue", "red2"))


```

```{r}

# hazard ratio, exp(coef) 
coxp4 <- coxph(Surv(time, status1) ~ quantile, data = prot_all4)
coxp4
```
## Survival analysis using clusters

```{r,fig.align='center',fig.height=7}

prot_h_clust <- cbind(t(prot_in), 
                     cluster = cutree(prot_h$tree_row, k = 2))
prot_h_clust <- as.data.frame(prot_h_clust)
prot_h_clust$case_id <- rownames(prot_h_clust)
p_clust_all <- merge(prot_h_clust,clinical_sub,by="case_id")
p_clust_all$status1 <- ifelse(p_clust_all$vital_status=="Deceased",1,0)

## Remove rows from p_clust_all which does not exist in surv_free file
p_clust_all <- p_clust_all[(p_clust_all$case_id %in% surv_free$Case), ]


fit_prot_clust <- survfit(Surv(time, status1) ~ cluster, data = p_clust_all)
  ggsurvplot(fit_prot_clust,
           pval = TRUE,
           ncensor.plot=FALSE,
           risk.table.col = "strata", 
           ggtheme = theme_bw(), 
           title = "Signature",
           palette = c("blue", "red2"))


```







